### PR TITLE
Adding more structure to Shoot Cluster Maintenance

### PR DESCRIPTION
**Pull request by JensHaley**
**What this PR does / why we need it**:

I could see that the structure was improved after PR #102, but I wanted to go further:
* clearer separation of examples and functional documentation, separate the WHAT from the HOW.
* More bullets, and tables to help readers get a better overview over the functionality

**Special notes for your reviewer**:

Since I moved paragraphs around it might be hard to review my changes, but believe me that all information is still there. Maybe it's best to read the [preview](https://github.com/JensHaley/documentation/blob/master/website/documentation/050-tutorials/content/howto/maintain-shoot/_index.md) first before starting the review in GitHub. **Thank you**.

There is only this change in section _Automatic Operating System Version Updates_ that I removed completely:

----

During the reconciliation, the corresponding <Provider>MachineClass resource in the Shoot namespace in the Seed will be updated and the machine controller manager will take care of the actual state to match the desired one.

----
The "machine controller manager" is not mentioned again in the whole document and to me it sounds as something the reader does not really have to know. So following "less is more", I removed it, but if you think it is really important information, I'll add it again. ;-)

I don't like the term _forceful update_ much (sounds a bit negative), but all other ideas I had didn't seem to be better (_planned update_, _updates based on expiration dates_, _profile-triggered updates_), so I kept it.

### CODE REVIEW

File: website/documentation/050-tutorials/content/howto/maintain-shoot/_index.md [hunk 1]
```diff
@@ -11,94 +11,154 @@ aliases: ["readmore/shoot-maintain"]
...
+</p>
 {{% /notice %}}
 
+With expiration dates, administrators can give shoot cluster owners more time for testing before the actual version update happens, which allows smoother transitions to new versions.
 
-### Forceful Version Updates
+### Kubernetes Update Paths
 
-While the [automatic Kubernetes](#kubernetes-automatic-version-updates) and [operating system updates](#automatic-operating-system-version-updates), are an opt-in feature for the Shoot cluster owner, Gardener administrators can force patch and minor version updates.
+The bigger the delta of the Kubernetes source version and the Kubernetes target version, the better it must be planned and executed by operators. Gardener only provides automatic support for updates that are uncritical for your deployed resources:
```
Comment (ialidzhikov) [inline website/documentation/050-tutorials/content/howto/maintain-shoot/_index.md:93]: critical != harmful, patch update can be critical (for example includes CVE fixes).

```suggestion
The bigger the delta of the Kubernetes source version and the Kubernetes target version, the better it must be planned and executed by operators. Gardener only provides automatic support for updates that are not harmful for the cluster workload:
```

File: website/documentation/050-tutorials/content/howto/maintain-shoot/_index.md [hunk 2]
```diff
@@ -11,94 +11,154 @@ aliases: ["readmore/shoot-maintain"]
 
 # Shoot Cluster Maintenance
 
-Day two operations like updating the Kubernetes patch version and updating the Operating system version, happen in a daily maintenance time window of the Shoot cluster. The maintenance time window is part of the shoot spec (`.spec.maintenance.timeWindow`). If it is not specified during Shoot creation, Gardener will default to a randomized time window (to spread the load). The time interval cannot be less than 30 minutes and more than 6 hours.
+Day two operations for shoot clusters are related to:
 
-When referring to an `operating system update` in this document, this means updating the Shoot's worker nodes with a machine image (e.g. [AMI](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/AMIs.html) for AWS) that comes with a higher version of the operating system.
+* The Kubernetes version of the control planes and the worker nodes and
```
Comment (ialidzhikov) [inline website/documentation/050-tutorials/content/howto/maintain-shoot/_index.md:16]:
```suggestion
* the Kubernetes version of the control plane and the worker nodes
```

File: website/documentation/050-tutorials/content/howto/maintain-shoot/_index.md [hunk 3]
```diff
@@ -11,94 +11,154 @@ aliases: ["readmore/shoot-maintain"]
...
+<p>
+The administrator who maintains the `CloudProfile` has to ensure that the list of Kubernetes versions consists of consecutive minor versions, for example, from `1.10.x` to `1.11.y`. If the minor version increases in bigger steps, for example, from `1.10.x` to `1.12.y`, shoot clusters updates fail during the maintenance window. 
+</p>
+{{% /notice %}}
+
+## Manual Updates
+
+To update the Kubernetes version or the node operating system manually, change the `.spec.kubernetes.version` field or the `.spec.provider.workers.machine.image.version` field correspondingly.  
+
+Manual updates are required if you would like to do a minor update or a major update of the Kubernetes version. Gardener doesn’t do such updates automatically as they can have breaking changes that could impact your deployed resources.
```
Comment (ialidzhikov) [inline website/documentation/050-tutorials/content/howto/maintain-shoot/_index.md:113]:
```suggestion
Manual updates are required if you would like to do a minor update of the Kubernetes version. Gardener doesn’t do such updates automatically as they can have breaking changes that could impact the cluster workload.
```

File: website/documentation/050-tutorials/content/howto/maintain-shoot/_index.md [hunk 4]
```diff
@@ -11,94 +11,154 @@ aliases: ["readmore/shoot-maintain"]
...
+spec:
+  kubernetes:
+    versions:
+    - version: 1.11.0
+    - version: 1.10.5
+    - version: 1.10.0
+```
 
-Let's assume the following CloudProfile spec (only related fields are shown):
+Before this change, your shoot cluster specification looked like this:
```
Comment (ialidzhikov) [inline website/documentation/050-tutorials/content/howto/maintain-shoot/_index.md:143]:
```suggestion
Before this change, the shoot cluster specification looked like this:
```

File: website/documentation/050-tutorials/content/howto/maintain-shoot/_index.md (hunk 1 above)
Comment (JensHaley) [reply website/documentation/050-tutorials/content/howto/maintain-shoot/_index.md:93]: @ialidzhikov: Thanks for your review (working on it)!

"harmful" does sound weird (I checked with a native speaker), but I see your point.

Proposal:

Gardener only provides automatic support for updates that can be applied safely to the cluster workload.

Other idea: ...that can be applied without disrupting the cluster workload...

File: website/documentation/050-tutorials/content/howto/maintain-shoot/_index.md [hunk 5]
```diff
@@ -11,94 +11,154 @@ aliases: ["readmore/shoot-maintain"]
...
+</p>
 {{% /notice %}}
 
+With expiration dates, administrators can give shoot cluster owners more time for testing before the actual version update happens, which allows smoother transitions to new versions.
 
-### Forceful Version Updates
+### Kubernetes Update Paths
 
-While the [automatic Kubernetes](#kubernetes-automatic-version-updates) and [operating system updates](#automatic-operating-system-version-updates), are an opt-in feature for the Shoot cluster owner, Gardener administrators can force patch and minor version updates.
+The bigger the delta of the Kubernetes source version and the Kubernetes target version, the better it must be planned and executed by operators. Gardener only provides automatic support for updates that can be applied safely to the cluster workload:
```
Comment (JensHaley) [inline website/documentation/050-tutorials/content/howto/maintain-shoot/_index.md:93]: Is "can be applied safely" o.k.?

**Comment by ialidzhikov** [comment]
Thank you @JensHaley . Can you resolve the conflict and squash to single commit?

**Comment by JensHaley** [comment]
@ialidzhikov : I cannot resolve the conflict yet myself, because I don't have write access to github.com/gardener/documentation. 
@msohn : Could you grant write access to me for github.com/gardener/documentation ?

**Pull request merged**